keps/2349-multikueue-external-custom-job-support/README.md

```diff
@@ -0,0 +1,151 @@
+
+### User Stories ( Optional )
+Ability to schedule resources not yet supported by kueue like Tekton.
+
+### Constraints
+
+- Limited Transformation: Generic Adaptor are limited to simple JsonPatch transformation configurable.
+
+### Risks and Mitigations
+
+#### Risks
+
+ Configmaps are unstructured by default. They don't provide validation by default unlike CRD. User could make typo mistakes or give invalid syntax in configmap. Or user could give invalid syntax.
+
+#### Mitigations
+
+Configmap which aren't correct should logged like Invalid GVK format, Invalid JsonPatch ops. We could have cli command to validate adapter logic.
+
+## Design Details
+
+### Generic Adapter via ConfigMap
+
+The `kueue-manager-config` configmap will be expanded to configure the generic adapter.
+
+```
+     # In the main Kueue ConfigMap
+     integrations:
+       externalFrameworks:
+         # OLD FORMAT
+         - "taskrun.tekton.dev"
+         # NEW, EXPANDED FORMAT (for MultiKueue)
+         - group: "pipelinerun.tekton.dev"
+           version: "v1"
+           kind: "PipelineRun"
+           multiKueueAdapter:
+            managedBy: ".spec.managedBy"
+            creationPatches:
+            # Used to define what to patch while creating in worker. // By default, we can assume replace with null for managedBy field.
+                - op: replace
+                  path: /spec/managedBy
+                  value: null
+            syncPatches:
+            # Copy the entire status block from the remote object to the local one. By default, we can assume /status. Unless some other fields also need to be copied.
+                - op: replace
+                  path: /status
+                  from: /status
+
+```
+
+       The  workload reconciler will be modified so that when it encounters a workload whose owner GVK does not match any of the built-in adapters, it will:
+        -  Look up the GVK from the configmap..
+        - If no entry is found, the workload is rejected.
+        - If an entry is found, it will use a generic adapter that performs actions based on the rules:
+        - *~IsJobManagedByKueue~*: Reads the local job object and uses the ~managedBy~ JSONPath to verify it's managed by MultiKueue.
+        - *~SyncJob~ (Create)*: Clones the local job object, applies the ~creationPatches~, adds the standard MultiKueue labels (~prebuilt-workload~, ~multikueue-origin~), and creates the resulting object
+       on the worker cluster.
+           - *~SyncJob~ (Status Sync)*: Fetches the local and remote job objects. It then iterates through the ~statusSyncPatches~, applying each one to construct a patch for the local job's status
+       subresource.
+
+### Workload-Only Sync for External Controllers
```

Comment:
@gbenhaim I think you are talking about this implementation.

Reply:
MultiKueue ACC can sync workload CR but to sync job CR, you need to have a way to define how to sync. Isn't that how the first approach is working?